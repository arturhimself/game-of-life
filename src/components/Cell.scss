.cell {
  position: relative;
  width: 10%;
  transition: .2s;
  cursor: pointer;

  &::before {
    content: '';
    display: block;
    padding-top: 100%;
  }

  border-right: 1px solid #c1c1c1;
  border-bottom: 1px solid #c1c1c1;

  &:hover {
    background: #ececec;
  }

  &._active {
    background: #ececec;
  }

  span {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
}