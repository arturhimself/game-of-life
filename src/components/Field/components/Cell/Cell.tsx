import React from 'react';
import { CellItem } from "./CellItems";

interface CellProps { 
  index?: number,
  filled: number,
  width: number,
}

const Cell: React.FC<CellProps> = ({ filled, width }) => {
  return (
    <CellItem width={width} isFilled={filled} />
  )
};

export { Cell };