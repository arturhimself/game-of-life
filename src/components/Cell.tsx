import { useState } from 'react';
import './Cell.scss';

interface CellProps { 
  index?: number,
  filled: number,
  styles?: Object,
}

const Cell: React.FC<CellProps> = ({ filled, styles }) => {
  return (
    <div style={styles} className={`cell ${filled ? '_active' : ''}`} />
  )
};

export default Cell;