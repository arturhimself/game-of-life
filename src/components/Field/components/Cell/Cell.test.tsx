import React from 'react';
import renderer from 'react-test-renderer';
import { Cell } from './Cell';

describe('Cell', () => {
  const width = 50;

  it('render empty cell', () => {
    expect(
      renderer.create(<Cell filled={0} width={width} />).toJSON()
    ).toMatchSnapshot();
  });

  it('render filled cell', () => {
    expect(
      renderer.create(<Cell filled={1} width={width} />).toJSON()
    ).toMatchSnapshot();
  });
});