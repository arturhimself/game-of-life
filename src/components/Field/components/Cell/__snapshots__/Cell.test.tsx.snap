// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell render empty cell 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  cursor: pointer;
  border-right: 1px solid #c1c1c1;
  border-bottom: 1px solid #c1c1c1;
  width: 50px;
}

.emotion-0::before {
  content: "";
  display: block;
  padding-top: 100%;
}

.emotion-0:hover {
  background: #ececec;
}

.emotion-0 span {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<span
  className="emotion-0"
  width={50}
/>
`;

exports[`Cell render filled cell 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  cursor: pointer;
  border-right: 1px solid #c1c1c1;
  border-bottom: 1px solid #c1c1c1;
  width: 50px;
  background: #000;
}

.emotion-0::before {
  content: "";
  display: block;
  padding-top: 100%;
}

.emotion-0:hover {
  background: #ececec;
}

.emotion-0 span {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<span
  className="emotion-0"
  width={50}
/>
`;
